### PR TITLE
drhuffman12 bump crystal-lang version to 1.0

### PR DESCRIPTION
**Pull request by drhuffman12**
Bump version of Crystal up to 1.0 (using the `crystallang/crystal:nightly-alpine-build` docker image).

Make runnable with Crystal 1.0.
(Some logic adjustments might be needed for getting 'correctly format 1.0 code', but you should at least be able to run Ameba with a Crystal-1.0-based app.)

### CODE REVIEW

**Comment by drhuffman12** [comment]
@veelenga , please review.

**Comment by Sija** [comment]
Most of the changes are unrelated, and the version should be changed to `>= 0.35` to make it work with all crystal releases.

**Comment by Sija** [comment]
@drhuffman12 You will probably run into the same issues with other shards btw.

**Comment by drhuffman12** [comment]
@veelenga / @Sija , Thanks for the input.

I think travis is stuck. Can you please restart https://travis-ci.org/github/crystal-ameba/ameba/builds/746902717 re https://github.com/crystal-ameba/ameba/pull/173/commits/aa00bfaafaf0f2ee2c70ebd9080527363f1ad5b3 . I'd like to know if that passes before I add any other changes.

File: shard.yml [hunk 1]
```diff
@@ -1,7 +1,8 @@
 name: ameba
-version: 0.13.2
+version: 0.13.3
+
+crystal: 1.0
```
Comment (veelenga) [inline shard.yml:4]: Is there any breaking change in Ameba which doesn't support crystal 0.29.0 now? I think here the idea is to specify the latest compatible crystal version (which is 0.29.0 in my understanding)

**Comment by veelenga** [comment]
@drhuffman12 thanks for your effort. The Travis build has passed.

File: shard.yml (hunk 1 above)
Comment (Sija) [reply shard.yml:4]: IIRC minimum supported version is `0.34` or `0.35`

**Comment by Sija** [comment]
Superseded by #174

**Pull request closed without merging**